service/metrics.go

```diff
@@ -0,0 +1,45 @@
+package service
+
+import (
+	"runtime/debug"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// MustRegister will register all metrics on the given registry.
+func MustRegister(registry *prometheus.Registry) {
+	registry.MustRegister(buildInfo)
+}
+
+// SampleBuildInfo creates a sample of the service_build_info metric.
+// Since it is a gauge it needs to be set only once on the service startup.
+func SampleBuildInfo() {
+	goVersion := "undefined"
+	revision := "undefined"
+
+	goBuildInfo, ok := debug.ReadBuildInfo()
+	if ok {
+		goVersion = goBuildInfo.GoVersion
+		for _, buildSetting := range goBuildInfo.Settings {
+			if buildSetting.Key == "vcs.revision" {
+				revision = buildSetting.Value
```

Comment:
This is the go revision, right? Shouldn't we also add our service version?

Reply:
not it is not, this is our service version (from vcs/git). The go version is goBuildInfo.GoVersion.

Reply:
Reference:https://pkg.go.dev/runtime/debug#BuildSetting

"vcs.revision: the revision identifier for the current commit or checkout"

Reply:
Mmh, shit! I always injected in the makefile, I didn't know about that.

Reply:
All good :-), this is new stuff that came on Go 1.18 (which now it is not that new actually...getting old sux :rofl: )